test/models/query.spec.js

```diff
@@ -0,0 +1,130 @@
+import chai from 'chai';
+
+import Query from '../../src/models/query';
+import QueryAggregate from '../../src/models/queryAggregate';
+
+describe('Query', function() {
+  describe('Construct', function() {
+    it('should create a basic object with empty fields, aggregates, filters and sort', function() {
+      const queryName = 'test_query';
+      const query = new Query(queryName);
+      chai.expect(query.name).to.equal(queryName); // eslint-disable-line
+      chai.expect(query.getFilters()).to.be.empty; // eslint-disable-line
+      chai.expect(query.getAggregates()).to.be.empty; // eslint-disable-line
+    });
+  });
+  describe('setFilters', function() {
+    it('should set a filter object on the query', function() {
+      const queryName = 'test_query';
+      const query = new Query(queryName);
+      const filters = {and: []};
+      query.setFilters(filters);
+      chai.expect(query.getFilters()).to.deep.equal(filters);
+    });
+  });
+  describe('addAggregate', function() {
+    it('should add a new instance of QueryAggregate', function() {
+      const queryName = 'test_query';
+      const query = new Query(queryName);
```

Comment:
this isn't the contructor signature anymore. to do pass any params
same every where.

test must reflect how you should use the lib.